src/WorldStorage/WSSAnvil.cpp

```diff
@@ -2663,15 +2665,20 @@ void cWSSAnvil::LoadSilverfishFromNBT(cEntityList & a_Entities, const cParsedNBT
 
 void cWSSAnvil::LoadSkeletonFromNBT(cEntityList & a_Entities, const cParsedNBT & a_NBT, int a_TagIdx)
 {
+	// Wither skeleton is a separate mob in Minecraft 1.11+, but we need this to
+	// load them from older worlds where wither skeletons were only a skeleton with a flag
 	int TypeIdx = a_NBT.FindChildByName(a_TagIdx, "SkeletonType");
-	if (TypeIdx < 0)
+
+	std::unique_ptr<cMonster> Monster;
+	if ((TypeIdx > 0) && (a_NBT.GetByte(TypeIdx) == 1))
 	{
-		return;
+		Monster.reset(new cWitherSkeleton);
```

Comment:
I thought make_unique supported upcasting?

Reply:
Why we use `new` + reset instead of make_unique

Reply:
I thought `make_unique` could be used only in the initial assignment, but that doesn't make sense. @mathiascode Try `Monster = cpp14::make_unique<cWitherSkeleton>()` here.